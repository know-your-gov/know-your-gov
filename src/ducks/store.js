import { createStore, applyMiddleware, compose } from "redux";
import rootReducer from "./rootReducer";
import thunk from "redux-thunk";
import { reduxFirestore, getFirestore } from "redux-firestore";
import { reactReduxFirebase, getFirebase } from "react-redux-firebase";
import fbConfig from "../config/fbConfig";

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(thunk.withExtraArgument({ getFirebase, getFirestore })),
    reduxFirestore(fbConfig),
    reactReduxFirebase(fbConfig, { attachAuthIsReady: true })
  )
);

export default store;
