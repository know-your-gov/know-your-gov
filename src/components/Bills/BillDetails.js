import React, { Component } from "react";
import axios from "axios";
// import Button from '@material-ui/core/Button'
import { connect } from "react-redux";
import { compose } from "redux";
import { firestoreConnect } from "react-redux-firebase";
import { billFavor } from "../../ducks/authReducer";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import FavoriteBorder from "@material-ui/icons/FavoriteBorder";
import Clear from "@material-ui/icons/Clear";
import CardActions from "@material-ui/core/CardActions";
import CardActionArea from "@material-ui/core/CardActionArea";
import { Link } from "react-router-dom";

class BillDetails extends Component {
  constructor() {
    super();
    this.state = {
      billDetails: {}
    };
  }
  componentDidMount = () => {
    this.getBillDetails();
  };
  getBillDetails = () => {
    const { billId } = this.props.match.params;
    const split = billId.split("-");
    const congress = split[1];
    const billSlug = split[0];

    axios
      .get(
        `https://api.propublica.org/congress/v1/${congress}/bills/${billSlug}.json`,
        {
          headers: { "X-API-Key": process.env.REACT_APP_PRO_PUBLICA }
        }
      )
      .then(res => {
        this.setState({ billDetails: res.data.results[0] });
      });
  };
  render() {
    const { billDetails } = this.state;
    console.log(billDetails);
    return (
      <div style={{ width: "65%", margin: "0 auto", marginTop: "5vh" }}>
        <Card>
          <CardContent>
            <Typography variant="h6">{billDetails.title}</Typography>
            <p>{billDetails.committees}</p>
            <p>
              Sponsored by:{" "}
              <Link to={`/politicians/${billDetails.sponsor_id}`}>
                {billDetails.sponsor_title} {billDetails.sponsor} (
                {billDetails.sponsor_party}, {billDetails.sponsor_state})
              </Link>
            </p>
            <p>Introduced(yyyy/mm/dd): {billDetails.introduced_date}</p>
            <p>
              {" "}
              <a
                href={billDetails.gpo_pdf_uri}
                target="_blank"
                rel="noopener noreferrer"
              >
                See more information here
              </a>
            </p>
            <div>
              <CardActionArea>
                <CardActions>
                  <FavoriteBorder
                    onClick={() => this.props.billFavor(billDetails.bill_id)}
                  />
                  <Clear />
                </CardActions>
              </CardActionArea>
            </div>
          </CardContent>
        </Card>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    auth: state.firebase.auth
  };
};
const mapDispatchToProps = dispatch => {
  return {
    billFavor: billId => dispatch(billFavor(billId))
  }; /////////////////////
};
export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  firestoreConnect(state => {
    return [
      { collection: "users", doc: state.auth.uid, collection: "bills-liked" }
    ];
  })
)(BillDetails);

/*
  // const congSenId = /(?<=-)\d{1,6}/
    // const congress = congSenId.exec(billId)[0]
    // const billSlug = billId.split(congSenId)[0].split("-")[0]

*/
