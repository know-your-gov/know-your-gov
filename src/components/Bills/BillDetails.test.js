import React from 'react'
import renderer from 'react-test-renderer'
import BillDetails from './BillDetails'
import {Provider} from 'react-redux'
import store from '../../ducks/store'
import {BrowserRouter as Router} from 'react-router-dom'


test("Bill Details page renders",()=>{
  const component = renderer.create(
  <Provider store = {store}>
    <Router>
      <BillDetails match = {{params:{billId:"h234-115"}}}/>
    </Router> 
  </Provider>
  )
  expect(component.toJSON()).toMatchSnapshot();
})
