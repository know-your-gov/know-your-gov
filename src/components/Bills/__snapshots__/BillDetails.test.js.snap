// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bill Details page renders 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
      "marginTop": "5vh",
      "width": "65%",
    }
  }
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
  >
    <div
      className="MuiCardContent-root-29"
    >
      <h6
        className="MuiTypography-root-30 MuiTypography-h6-47"
      />
      <p />
      <p>
        Sponsored by:
         
        <a
          href="/politicians/undefined"
          onClick={[Function]}
        >
           
           (
          , 
          )
        </a>
      </p>
      <p>
        Introduced(yyyy/mm/dd): 
      </p>
      <p>
         
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          See more information here
        </a>
      </p>
      <div>
        <button
          className="MuiButtonBase-root-69 MuiCardActionArea-root-66"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="MuiCardActions-root-72"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-75 MuiCardActions-action-74"
              focusable="false"
              onClick={[Function]}
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-75 MuiCardActions-action-74"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight-68"
          />
          <span
            className="MuiTouchRipple-root-84"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
